pkg/controller/master/image/vm_image_controller.go

```diff
@@ -127,16 +127,26 @@ func (h *vmImageHandler) createBackingImage(image *harvesterv1.VirtualMachineIma
 		bi.Spec.SourceParameters[types.DataSourceTypeDownloadParameterURL] = image.Spec.URL
 	}
 
+	if image.Spec.SourceType == harvesterv1.VirtualMachineImageSourceTypeExportVolume {
+		bi.Spec.SourceParameters[types.DataSourceTypeExportFromVolumeParameterVolumeName] = image.Spec.VolumeName
+		bi.Spec.SourceParameters[types.DataSourceTypeExportFromVolumeParameterExportType] = types.DataSourceTypeExportFromVolumeParameterExportTypeRAW
+	}
+
 	_, err := h.backingImages.Create(bi)
 	return err
 }
 
 func (h *vmImageHandler) createStorageClass(image *harvesterv1.VirtualMachineImage) error {
+	// we don't need to create storageclass if we are exporting existing volume to image
+	if image.Spec.SourceType == harvesterv1.VirtualMachineImageSourceTypeExportVolume {
+		return nil
+	}
```

Comment:
We should create a storageclass as well. A VMImage type volume of a VM is a PVC with the storageclass of the image.

Reply:
@gitlawr  updated